### PR TITLE
feat: new delete message

### PR DESCRIPTION
**Pull request by yqrashawn**
Related: #14188
Corresponding status-go pr https://github.com/status-im/status-go/pull/2922

### Summary

add undo logic to delete similar to delete-for-me

* [x] undo logic
* [x] display deleted messages logic
* [x] deleted messages UI
* [x] reply deleted message UI https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=10364%3A123939&t=KxTsDEHlb7o8WlHm-4
* [x] unpin when deleted for real and has the permission to unpin
* [x] don't allow delete for me for pinned messages
* [ ] delete pinned message when don't have perms to unpin the message https://github.com/status-im/status-mobile/issues/14488
* [ ] undo deletion toast UI (will implement this in separate PR) https://github.com/status-im/status-mobile/pull/14376
* [ ] deleted by xxx UI, may need to add `deleted-by` in status-go https://github.com/status-im/status-mobile/issues/14489

### Testing notes

- Deleted message won't disappear, but shown as a system message indicating this is a deleted message.
- Will only broadcast delete action after 4s of delay

status: ready

### CODE REVIEW

File: src/status_im/chat/models/input.cljs [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
             [status-im.constants :as constants]
+            [status-im.ethereum.json-rpc :as json-rpc]
             [status-im.i18n.i18n :as i18n]
             [status-im.utils.datetime :as datetime]
             [status-im.utils.fx :as fx]
-            ["emojilib" :as emojis]
-            [status-im.chat.models.mentions :as mentions]
-            [status-im.utils.utils :as utils]))
+            [status-im.utils.utils :as utils]
+            [taoensso.timbre :as log]))
```
Comment (yqrashawn) [inline src/status_im/chat/models/input.cljs:17]: reorder only

File: src/status_im/chat/models/input.cljs [hunk 2]
```diff
@@ -133,16 +133,6 @@
...
-(fx/defn soft-delete-message
-  "Does't delete from db, this is a soft delete"
-  {:events [:chat.ui/soft-delete-message]}
-  [{:keys [db] :as cofx} {:keys [message-id chat-id]}]
-  {::json-rpc/call [{:method      "wakuext_deleteMessageAndSend"
-                     :params      [message-id]
-                     :js-response true
-                     :on-error    #(log/error "failed to delete message message " %)
-                     :on-success  #(re-frame/dispatch [:sanitize-messages-and-process-response %])}]})
-
```
Comment (yqrashawn) [inline src/status_im/chat/models/input.cljs:146]: use separate ns

File: src/status_im/chat/models/message.cljs [hunk 3]
```diff
@@ -183,7 +183,7 @@
 
 (fx/defn delete-message
   "Deletes chat message, rebuild message-list"
-  {:events [:chat.ui/delete-message]}
+  {:events [:chat.ui/delete-message-not-used-any-more]}
```
Comment (yqrashawn) [inline src/status_im/chat/models/message.cljs]: This is not used anymore

File: src/status_im/chat/models/message.cljs [hunk 4]
```diff
@@ -208,10 +208,7 @@
                                                                           [message-id]
                                                                           #(re-frame/dispatch [:chat/decrease-unviewed-count chat-id %3])))) removed-messages)
         remove-messages-fx (fn [{:keys [db]}]
-                             {:db (reduce (fn [acc current]
-                                            (update-in acc [:messages (:chatId current)] dissoc (:messageId current)))
-                                          db removed-messages)
-                              :dispatch-n [[:get-activity-center-notifications]
```
Comment (yqrashawn) [inline src/status_im/chat/models/message.cljs:214]: So that deleted messages won't get removed.

File: src/status_im/ui/screens/chat/sheets.cljs [hunk 5]
```diff
@@ -177,7 +177,7 @@
        :title               (i18n/label :t/delete-message)
        :icon                :main-icons/delete
        :accessibility-label :delete-transaccent-button
-       :on-press            #(hide-sheet-and-dispatch [:chat.ui/delete-message chat-id message-id])}]]))
+       :on-press            #(hide-sheet-and-dispatch [:chat.ui/delete-message-not-used-any-more chat-id message-id])}]]))
```
Comment (yqrashawn) [inline src/status_im/ui/screens/chat/sheets.cljs:180]: Not in ui2.

File: src/status_im/ui2/screens/chat/messages/message.cljs [hunk 6]
```diff
@@ -277,17 +278,20 @@
...
       [system-message/system-message
        {:type             :deleted
-        :label            :message-deleted-for-you
+        :label            (if deleted? :message-deleted :message-deleted-for-you)
         :timestamp-str    timestamp-str
         :non-pressable?   true
-        :animate-landing? (if deleted-for-me-undoable-till true false)}]
+        :animate-landing? (if (or deleted-undoable-till deleted-for-me-undoable-till)
+                            true
+                            false)}]
```
Comment (yqrashawn) [inline src/status_im/ui2/screens/chat/messages/message.cljs]: Deleted and deleted-for-me use the same `system-message` type, only difference is the label

File: src/status_im/chat/models/delete_message.cljs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+(fx/defn delete-and-send
+  {:events [:chat.ui/delete-message-and-send]}
+  [{:keys [db]} {:keys [message-id chat-id]}]
+  (when (get-in db [:messages chat-id message-id])
+    {:db             (update-db-clear-undo-timer db chat-id message-id)
+     ::json-rpc/call [{:method      "wakuext_deleteMessageAndSend"
+                       :params      [message-id]
+                       :js-response true
+                       :on-error    #(log/error "failed to delete message " %)
```
Comment (ilmotta) [inline src/status_im/chat/models/delete_message.cljs]: I've recently started to think a bit more about error handling in the Status mobile app.

One of the first things I do is evaluate what's the necessary data that should be logged in order to improve _observability_. In this particular case @yqrashawn, it could be a good idea to log the message ID.

One other point of view is, can we trust the backend is logging all the necessary data in case an RPC call fails? What information is contained in the RPC error object? Is the error object complete enough in case an error occurs? If it's complete enough, we might not need to log anything at all in the frontend. There are tons of stuff to review for error handling, it'd be too boring to discuss that over here. Nonetheless, I'm leaving a comment simply to raise awareness in the team.

File: src/status_im/chat/models/delete_message_test.cljs [hunk 8]
```diff
@@ -0,0 +1,92 @@
+(ns status-im.chat.models.delete-message-test
+  (:require
+   [cljs.test                            :refer-macros [deftest is testing]]
+   [status-im.chat.models.delete-message :as delete-message]
+   [status-im.utils.datetime             :as datetime]))
+
+(defonce mid "message-id")
```
Comment (ilmotta) [inline src/status_im/chat/models/delete_message_test.cljs]: `defonce` is often used for stateful data, is there a particular reason you chose to use `defonce` instead of just `def`?

File: src/status_im/chat/models/delete_message_test.cljs [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  (with-redefs [datetime/timestamp (constantly 1)]
+    (let [db      {:messages {cid {mid {:id mid :whisper-timestamp 1}}}}
+          message {:message-id mid :chat-id cid}]
+      (testing "delete"
+        (let [result-message (get-in (delete-message/delete {:db db} message 1000)
+                                     [:db :messages cid mid])]
+          (is (= (:id result-message) mid))
+          (is (true? (:deleted? result-message)))
+          (is (= (:deleted-undoable-till result-message) 1001))))
+      (testing "should return nil if message not in db"
```
Comment (ilmotta) [inline src/status_im/chat/models/delete_message_test.cljs]: For consistency sake, something most teams agree early on is on how tests descriptions should be written. Given we are in the early days of increasing test coverage, you could try to be consistent in the namespace level.

My own convention is to write something actionable, in the 3rd person, usually starting with a verb. I usually avoid the `should` word because it ends up being superfluous, because one could say that every test _should do something_. If you prefer to use the word should, you could reword tests to always use them, etc, it's up to you of course. In the future, and with more tests, we can standardize a bit more.

Examples translated for your PR:

- `returns nil if message not in db`
- `remains deleted when undo delete late`
- etc

File: src/status_im/chat/models/delete_message_test.cljs [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+    (testing "remain deleted when undo delete late"
+      (let [db             (update-in db
+                                      [:messages cid mid]
+                                      assoc
+                                      :deleted?              true
+                                      :deleted-undoable-till (- (datetime/timestamp) 1000))
+            result-message (get-in (delete-message/undo {:db db} message) [:db :messages cid mid])]
+        (is (= (:id result-message) mid))
+        (is (nil? (:deleted-undoable-till result-message)))
```
Comment (ilmotta) [inline src/status_im/chat/models/delete_message_test.cljs]: Do we really need to assert against `nil?` or `true?`? It's not wrong in any way, but In Clojure, Ruby and other languages, it's very common to take advantage of truthy/falsey values because we don't need to check explicitly for types. For example:

```clojure
(is (not (:deleted-undoable-till result-message)))
(is (:deleted? result-message))
```

File: src/status_im/chat/models/delete_message.cljs [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+       (conj inner-acc [:chat.ui/delete-message-and-send chat-id message-id])
+       inner-acc))
+   acc
+   messages))
+
+(fx/defn send-all
+  "Get all deleted messages that not yet synced with status-go and send them"
+  {:events [:chat.ui/send-all-deleted-messages]}
+  [{:keys [db]}]
+  {:dispatch-n (reduce-kv chats-reducer [] (:messages db))})
```
Comment (ilmotta) [inline src/status_im/chat/models/delete_message.cljs]: I'm sharing this snippet just in case you like it because I found the reduce-kv with another inner reduce-kv call a bit difficult to understand. When I see a reduce call with a filtering step, I think: _Can we decompose it as a `filter + map` and would that improve readability?_ For the trained Clojure eye, a `filter + map` tells a story without even having to scan the code, whereas the reducer function requires a more thorough scan. Please, forgive me if I got your implementation wrong ;)

```clojure
(fx/defn send-all
  {:events [:chat.ui/send-all-deleted-messages]}
  [{:keys [db]}]
  {:dispatch-n (->> db
                    :messages
                    vals
                    (filter #(and (:deleted? %) (:deleted-undoable-till %)))
                    (mapv #(vector :chat.ui/delete-message-and-send chat-id (:id %))))})
```

**Review by ilmotta — APPROVED** [review]

**Review by ilmotta — COMMENTED** [review]
I left a bunch of comments that I found valuable, but PR LGTM :rocket:

**Comment by pavloburykh** [comment]
@yqrashawn thanx for the PR. could you please clarify if PR is ready for manual testing? 

Currently I don't see any of UI changes that should be introduced by this PR. 

### ISSUE 1 Unable to undo delete action

**Steps:** 

1. Start any chat
2. Send a message
3. Delete a message
4. See if it is possible to undo delete action

**Actual result:** undo element is not displayed. User is unable to undo delete action. 

**Expected result:** element with timer and Undo button is not displayed after message deletion 


https://user-images.githubusercontent.com/97245802/199023332-7e88ade8-a8e9-4ba6-a858-fa118ba2f158.mp4

**Comment by pavloburykh** [comment]
@yqrashawn also, could you please rebase PR to the latest develop? Thanx

**Comment by yqrashawn** [comment]
@pavloburykh not yet, waiting for the status-go pr https://github.com/status-im/status-go/pull/2922
The status-go changes not included in this PR yet.

**Comment by pavloburykh** [comment]
> @pavloburykh not yet, waiting for the status-go pr [status-im/status-go#2922](https://github.com/status-im/status-go/pull/2922) The status-go changes not included in this PR yet.

oh, got, thanx! Please, ping me up when it will be ready for manual testing.

File: src/status_im/chat/models/delete_message.cljs (hunk 7 above)
Comment (yqrashawn) [reply src/status_im/chat/models/delete_message.cljs]: > it could be a good idea to log the message ID
> can we trust the backend is logging all the necessary data in case an RPC call fails? What information is contained in the RPC error object? 

There's definitely no message ID in RPC error message if passing a wrong type of message ID (JSON unmarshal error), so better add message ID here

File: src/status_im/chat/models/delete_message.cljs (hunk 11 above)
Comment (yqrashawn) [reply src/status_im/chat/models/delete_message.cljs]: Done

File: src/status_im/chat/models/delete_message_test.cljs (hunk 9 above)
Comment (yqrashawn) [reply src/status_im/chat/models/delete_message_test.cljs]: Done

File: src/status_im/chat/models/delete_message_test.cljs (hunk 8 above)
Comment (yqrashawn) [reply src/status_im/chat/models/delete_message_test.cljs]: Done

File: src/status_im/chat/models/delete_message_test.cljs (hunk 10 above)
Comment (yqrashawn) [reply src/status_im/chat/models/delete_message_test.cljs]: I prefer to assert these values against `nil?` and `true?`, since these values aren't supposed to be set directly by functions in other namespaces.

File: src/status_im/chat/models/delete_message_test.cljs (hunk 9 above)
Comment (ilmotta) [reply src/status_im/chat/models/delete_message_test.cljs]: Thanks!

File: src/status_im/chat/models/delete_message.cljs (hunk 7 above)
Comment (ilmotta) [reply src/status_im/chat/models/delete_message.cljs]: I see from recent commits you added the message ID to the log in case an error happens, so we're gold now ;)

----

The following is not a suggestion for change, but rather something you might like to use, just as I do. I usually prefer to log errors by passing a human readable string, but also an additional map to contextualize the error in a parseable way. The timbre lib allows (and even encourages) us to this. Of course there are tons of nuances for how a team decides what to log, but you get the idea. For example:

```clojure
#(log/error "Failed to delete message" {:message-id message-id
                                        :error      %})
```

**Comment by yqrashawn** [comment]
status-go pr merged, icon gone after a big rebase, will look into this tomorrow, convert to draft for now

**Comment by du82** [comment]
What's the point of the "Deleted message" thing in the chat? If I wanted to hide that I deleted a message, having a "Deleted message" notice in chat is the opposite of helpful. Please offer a way to disable this feature.

**Comment by yqrashawn** [comment]
@du64 good point, I'll let the designer know.
@pavloburykh ready for test

**Comment by pavloburykh** [comment]
@yqrashawn thanx for the PR. Please, take a look at the following issues

### ISSUE 1 Message is not displayed as deleted for the receiver if his chat screen was opened during message has been deleted by sender

**Steps:**
1. Start any chat between User A and User B
2. Keep chat screen opened by both users
3. Send some message by User A
4. Delete for everyone message by User A
5. See if message has been deleted for User B

**Actual result:** message remains undeleted for User B until he re-opens the chat
**Expected result:** message is displayed as deleted for User B without chat re-opening


https://user-images.githubusercontent.com/97245802/202691572-b8658497-d1c2-40ff-b1b8-1b5ceb9a4190.mp4

[sender_geth.log](https://github.com/status-im/status-mobile/files/10040746/sender_geth.log)
[sender_Status.log](https://github.com/status-im/status-mobile/files/10040747/sender_Status.log)
[receiver_geth.log](https://github.com/status-im/status-mobile/files/10040749/receiver_geth.log)
[receiver_Status.log](https://github.com/status-im/status-mobile/files/10040750/receiver_Status.log)

**Comment by pavloburykh** [comment]
### ISSUE 2 Reactions for the deleted message are not deleted together with the message

**Steps:**
1. Start any chat between User A and User B
3. Send some message by User A
4. Set reaction to the message (by any user)
5. Delete for everyone message by User A
6. See if reactions are also deleted together with the message

**Actual result:** reactions are deleted
**Expected result:** reactions element should be deleted together with the message


https://user-images.githubusercontent.com/97245802/202693437-dfa5973d-1b00-4288-943c-a4c96099e9ea.mp4

**Comment by pavloburykh** [comment]
@yqrashawn could you also please rebase the PR to the latest develop and resolve conflicts in src/status_im/utils/config.cljs ?

**Comment by yqrashawn** [comment]
@pavloburykh rebased and able to reproduce both issues on my side
I'll ping you once it's fixed

**Comment by pavloburykh** [comment]
> @pavloburykh rebased and able to reproduce both issues on my side I'll ping you once it's fixed

Thank you @yqrashawn! When PR will be ready for testing, please move it to the E2E column on the board.

**Comment by yqrashawn** [comment]
@pavloburykh ready for test

**Comment by pavloburykh** [comment]
@yqrashawn thanx for your fixes. 

[ISSUE 2](https://github.com/status-im/status-mobile/pull/14232#issuecomment-1319869102) is partially reproducible: reactions are still not deleted if message is Deleted for me. 


https://user-images.githubusercontent.com/97245802/203079908-1cb4884f-a44e-4d4b-ba8a-7f8fbd32fcaa.mp4

**Comment by yqrashawn** [comment]
> reactions are still not deleted if message is Deleted for me.

Oops, sorry for that, fixed with new push

**Comment by pavloburykh** [comment]
@yqrashawn thank you. I have found another issue, please take a look

### ISSUE 3 The last messages are not deleted if there are more than 1 deleted messages in a row

**Steps:**

1. Send more than 1 message (better 5) in any chat
2. Start deleting messages one after another
3. Re-open the chat
4. See if all messages are preserved as deleted

**Actual result:** the last 1 or 2 messages that have been deleted are displayed as undeleted

**Expected result:** all messages should remain deleted


https://user-images.githubusercontent.com/97245802/203297621-2340b76e-43ae-4c37-b1cf-e0729970ea81.mp4

[geth.log](https://github.com/status-im/status-mobile/files/10066704/geth.log)
[Status.log](https://github.com/status-im/status-mobile/files/10066706/Status.log)

**Comment by pavloburykh** [comment]
@yqrashawn PR has been automatically moved to TO REBASE column, because of existing conflicts. Could you please resolve conflicts and then move PR back to E2E column when it will be ready for testing. 

You can also find information regarding current PR workflow process here https://notes.status.im/M73LlV1gT3OJ8kdSMY-ZQQ?view

Thanx for your work and patience!

**Comment by yqrashawn** [comment]
@pavloburykh thanks, ready for test

**Comment by yqrashawn** [comment]
rebased again

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn. Thank you for your PR. Take a look the following issues. And let's wait on a reply from  @pedro-et side

**ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned**
**Steps to reproduce:**
1. Create 1-1 chat as User_A with User_B
2. Create a message within the chat
3. Pin this message
4.  Remove this message using 'Deleted for everyone' option
5. Reopen current chat
6. Go to the list of pinned messages
7. Try to unpin the removed message

**Actual result:**
- 'Deleted for everyone' message is shown within list of pinned messages
- Removed message can't be unpinned from the list of pinned messages

https://user-images.githubusercontent.com/52490791/204050348-ceab6054-5010-4549-8fb5-5c95505157c8.mp4


**Expected result:**
- The 'Delete for everyone' message is removed from the list of pinned messages 
- Removed message is able to be unpinned from the list of pinned messages

**ISSUE 5. 'Deleted for everyone' message is shown as deleted in the pinned list only after chat is reopened**

**Steps to reproduce:**
1. Create 1-1 chat as User_A with User_B
2. Create a message within the chat
3. Pin this message
4.  Remove this message using 'Deleted for everyone' option
5. Reopen the current chat

**Actual result:**
'Deleted for everyone' message is shown with the initial content unless the chat is reopened
 
https://user-images.githubusercontent.com/52490791/204246780-7b5b2cb9-c2c9-47e4-bbec-befc3c11b97a.mp4

**Expected result:**
'Deleted for everyone' message is shown as deleted


**ISSUE 6. 'Deleted for me' message is removed for both users within list of pinned message** 
**Steps to reproduce:**
1. Create 1-1 chat as User_A with User_B
2. Create message within chat
3. Pin this message
4.  Remove this message using 'Deleted for me' option

**Actual result:**
'Deleted for me' message is removed for both users
https://user-images.githubusercontent.com/52490791/204041475-7d5f989d-a91d-4ef5-8b2b-85abd1dba441.mp4


**Expected result:**
'Deleted for me' message is removed for only user who removed current message from the list of pinned messages

Hi @pedro-et take a look at the expected result of ISSUE 1 is it suitable behavior? I have investigated how it works within Telegram and WhatsApp services and the behavior is the following: if user removes the message as  'Delete for everyone, then, the current message disappears from the list of pinned for all users. Should it disappear in Status or be shown as removed? 

Currently, it disappears from the list of pinned messages if 'delete for me' option is used, but shown as deleted if 'delete for everyone'  option is used and then can't be unpinned
image
Also, take a look at ISSUE 2 in the expected result section. Is it correct behavior that the message should be removed from the list of pinned only for the user who removes the current message?

**Comment by VolodLytvynenko** [comment]
> Hi @yqrashawn. Thank you for your PR. Take a look the following issues. And let's wait on a reply from @pedro-et side
> 
> **ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen current chat
> 6. Go to the list of pinned messages
> 7. Try to unpin the removed message
> 
> **Actual result:**
> 
> * 'Deleted for everyone' message is shown within list of pinned messages
> * Removed message can't be unpinned from the list of pinned messages
> 
>  cantunpin.mp4 
> **Expected result:**
> 
> * The 'Delete for everyone' message is removed from the list of pinned messages
> * Removed message is able to be unpinned from the list of pinned messages
> 
> **ISSUE 5. 'Deleted for everyone' message is shown as deleted in the pinned list only after chat is reopened**
> 
> **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen the current chat
> 
> **Actual result:** 'Deleted for everyone' message is shown with the initial content unless the chat is reopened
> 
>  deleted.mp4 
> **Expected result:** 'Deleted for everyone' message is shown as deleted
> 
> **ISSUE 6. 'Deleted for me' message is removed for both users within list of pinned message** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create message within chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for me' option
> 
> **Actual result:** 'Deleted for me' message is removed for both users https://user-images.githubusercontent.com/52490791/204041475-7d5f989d-a91d-4ef5-8b2b-85abd1dba441.mp4
> 
> **Expected result:** 'Deleted for me' message is removed for only user who removed current message from the list of pinned messages
> 
> Hi @pedro-et take a look at the expected result of ISSUE 1 is it suitable behavior? I have investigated how it works within Telegram and WhatsApp services and the behavior is the following: if user removes the message as 'Delete for everyone, then, the current message disappears from the list of pinned for all users. Should it disappear in Status or be shown as removed?
> 
> Currently, it disappears from the list of pinned messages if 'delete for me' option is used, but shown as deleted if 'delete for everyone' option is used and then can't be unpinned image Also, take a look at ISSUE 2 in the expected result section. Is it correct behavior that the message should be removed from the list of pinned only for the user who removes the current message?

hi @pedro-et 
You may have missed this message over the weekend. Just a mention for you :)

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn should the showing of the deleted message be in the scope of the current PR?
**On PR:**
image

**On design:**
image

**Comment by yqrashawn** [comment]
> should the showing of the deleted message be in the scope of the current PR?

Yes, any already implemented related features should be covered. 
Do you have some test cases can share with me? @VladimrLitvinenko 

Also PR not ready for test yet. Just a rebase. I'll move this to e2e test once ready for test.

**Comment by VolodLytvynenko** [comment]
> > should the showing of the deleted message be in the scope of the current PR?
> 
> Yes, any already implemented related features should be covered. Do you have some test cases can share with me? @VladimrLitvinenko
> 
> Also PR not ready for test yet. Just a rebase. I'll move this to e2e test once ready for test.

sure. Here is a list of test cases. Let me know if you need some more details https://docs.google.com/document/d/1oi-QarOL86JdGGdFjnB-jFuPby6B2bDGyctZfUtt6Yg/edit?usp=sharing

**Comment by VolodLytvynenko** [comment]
> the



> Hi @yqrashawn. Thank you for your PR. Take a look the following issues. And let's wait on a reply from @pedro-et side
> 
> **ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen current chat
> 6. Go to the list of pinned messages
> 7. Try to unpin the removed message
> 
> **Actual result:**
> 
> * 'Deleted for everyone' message is shown within list of pinned messages
> * Removed message can't be unpinned from the list of pinned messages
> 
>  cantunpin.mp4 
> **Expected result:**
> 
> * The 'Delete for everyone' message is removed from the list of pinned messages
> * Removed message is able to be unpinned from the list of pinned messages
> 
> **ISSUE 5. 'Deleted for everyone' message is shown as deleted in the pinned list only after chat is reopened**
> 
> **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen the current chat
> 
> **Actual result:** 'Deleted for everyone' message is shown with the initial content unless the chat is reopened
> 
>  deleted.mp4 
> **Expected result:** 'Deleted for everyone' message is shown as deleted
> 
> **ISSUE 6. 'Deleted for me' message is removed for both users within list of pinned message** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create message within chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for me' option
> 
> **Actual result:** 'Deleted for me' message is removed for both users https://user-images.githubusercontent.com/52490791/204041475-7d5f989d-a91d-4ef5-8b2b-85abd1dba441.mp4
> 
> **Expected result:** 'Deleted for me' message is removed for only user who removed current message from the list of pinned messages
> 
> Hi @pedro-et take a look at the expected result of ISSUE 1 is it suitable behavior? I have investigated how it works within Telegram and WhatsApp services and the behavior is the following: if user removes the message as 'Delete for everyone, then, the current message disappears from the list of pinned for all users. Should it disappear in Status or be shown as removed?
> 
> Currently, it disappears from the list of pinned messages if 'delete for me' option is used, but shown as deleted if 'delete for everyone' option is used and then can't be unpinned image Also, take a look at ISSUE 2 in the expected result section. Is it correct behavior that the message should be removed from the list of pinned only for the user who removes the current message?

Hi @xAlisher take a look please at the question addressed to Pedro. Pedro is not available, maybe you can help us

**Comment by yqrashawn** [comment]
> Re. your questions:
> 1. yes, we allow user to delete a pinned message without unpinning first
> 2. yes, when a pinned message is deleted, at the same time it is deleted it should also be unpinned

John replied in mobile-ui channel @VladimrLitvinenko

**Comment by VolodLytvynenko** [comment]
> > Re. your questions:
> > 
> > 1. yes, we allow user to delete a pinned message without unpinning first
> > 2. yes, when a pinned message is deleted, at the same time it is deleted it should also be unpinned
> 
> John replied in mobile-ui channel @VladimrLitvinenko

Oh. I see, thank you

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn. fix please additional issues:
**ISSUE 7. The '3 random names'/ ens / Nickname is not shown of user owner who deleted the message using Deleted for everyone option**
**Steps to reproduce:**
1. Open 1-1 chat as USER_A and USER_B
2. Send some message as USER_A
3. Remove the message using 'Delete for everyone'

**Actual result:**
USER_B sees 
image

**Expected result:**
USER_B sees the 3 random name or ens or nickname of a USER_A
image
USER_A sees 
image

**ISSUE 8. 'Message deleted for everyone' should be shown instead of 'Message deleted'**
**Steps to reproduce:**
1. Open 1-1 chat 
2. Send some message 
3. Remove the message using 'Delete for everyone'

**Actual result:**
image

**Expected result:**
image

***ISSUE 9. The replied message should be shown as deleted**
**Steps to reproduce:**
1. Open 1-1 chat 
2. Send some message 
3. Reply on the message
5. Remove message that was sent in step 2

**Actual result:**
The message is not shown as removed
image

**Expected result:**
The message is shown as removed
image

**Comment by yqrashawn** [comment]
> > > should the showing of the deleted message be in the scope of the current PR?
> > 
> > 
> > Yes, any already implemented related features should be covered. Do you have some test cases can share with me? @VladimrLitvinenko
> > Also PR not ready for test yet. Just a rebase. I'll move this to e2e test once ready for test.
> 
> sure. Here is a list of test cases. Let me know if you need some more details [docs.google.com/document/d/1oi-QarOL86JdGGdFjnB-jFuPby6B2bDGyctZfUtt6Yg/edit?usp=sharing](https://docs.google.com/document/d/1oi-QarOL86JdGGdFjnB-jFuPby6B2bDGyctZfUtt6Yg/edit?usp=sharing)

🤩 Thanks for the test cases.

**Comment by VladimrLitvinenko** [comment]
HI @yqrashawn some updates from John's side, share our discussion from UI-mobile
- hi John_  thank you for answers.One more question. Should the message be unpinned for all users if 'delete for you' option is used? or only unpinned for current user?

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn updates from John's side that I wanted to share with you. I think this is not in the scope of this PR as UI is not ready in Figma yet
-  **Me:** John_ One more question. Should the message be unpinned for all users  if 'delete for you' option is used? or only unpinned for the current user?
- **John:** Good question! I think we might actually want to now display the 'delete for you' option for any message that's pinned, otherwise is a user who is for example the Owner of a community uses the 'delete for you' option for a pinned message, and the community doesn't have any other admins and if the 'only owners and admins can pin and unpin messages' setting is enabled, then there would be no easy way to unpin the message in the future.
So if we remove the 'delete for you' option from any message that is pinned we can avoid these nasty corner cases.  cc'ing Alisher

**Comment by yqrashawn** [comment]
> Hi @yqrashawn updates from John's side that I wanted to share with you. I think this is not in the scope of this PR as UI is not ready in Figma yet
> 
>     * **Me:** John_ One more question. Should the message be unpinned for all users  if 'delete for you' option is used? or only unpinned for the current user?
> 
>     * **John:** Good question! I think we might actually want to now display the 'delete for you' option for any message that's pinned, otherwise is a user who is for example the Owner of a community uses the 'delete for you' option for a pinned message, and the community doesn't have any other admins and if the 'only owners and admins can pin and unpin messages' setting is enabled, then there would be no easy way to unpin the message in the future.
>       So if we remove the 'delete for you' option from any message that is pinned we can avoid these nasty corner cases.  cc'ing Alisher

Thanks. I just pushed new commit to hide pinned deleted for me/everyone messages. Although not ready for test yet.

**Comment by yqrashawn** [comment]
@VladimrLitvinenko ready for test 

Since there are many edge cases about delete message for me/everyone, I think it's better to get this merged at the earliest so that devs will get the idea of this logic and raise/solve edge cases while working on new features.

There are 2 issues that will be resolved with future PRs. 

**ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned**
partially fixed, will now try to unpin the message (without checking if user has the permission to unpin message) when syncing deleted pinned message with status-go

**ISSUE 7. The '3 random names'/ ens / Nickname is not shown of user owner who deleted the message using Deleted for everyone option**
not implemented, might require to persist “deleted by” info in status-go

I've created follow-up issue for them 
https://github.com/status-im/status-mobile/issues/14488
https://github.com/status-im/status-mobile/issues/14489

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn . Thank you! Test results for current PR will be given soon

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn . Some issues were found

**ISSUE 10. The pinned message should not be able to be with 'delete for you' option**

**Steps to reproduce:**
1. Create 1-1 chat
2. Send the message as USER_A
3. Pin the message as USER_A
4. Tap on the message as USER_B

**Actual result:**
'Deleted for me' option is still shown
image

**Expected result:**
The 'delete for me option' is not shown for all users If message is pinned

**ISSUE 11. 'Deleted for everyone' option disappears from the chat if to reopen it**

**Steps to reproduce:**
1. Create 1-1 chat
2. Send the message
3. Remove this message using 'Deleted for everyone' option
4. Reopen the chat

**Actual result:**
'Delete for everyone' option is not shown in the chat

https://user-images.githubusercontent.com/52490791/205680865-0f71a8bf-95b1-4cdf-91b4-8a8381050d11.mp4


**Expected result:**
'Delete for everyone' option is shown 


**ISSUE 12. 'Replied message is not shown as deleted for USER_B**

**Steps to reproduce:**
1. Create message in 1-1 chat
2. Send the message as USER_A with text "Hello world 1"
3. Leave reply for current message as USER_A
4. Remove message from step2 as USER_A using 'delete for everyone option'
5. Check the message as USER_B

**Actual result:**
USER_B sees
image

**Expected result:**
USER_B sees that the replied message is deleted
image

**Comment by yqrashawn** [comment]
@VladimrLitvinenko issue 11 and 12 are fixed, ready for test again

**Comment by VolodLytvynenko** [comment]
hi @yqrashawn thank you for your work. Current PR is tested and ready to be merged. All edge cases and cases related with community will be added to the separate PRs

**Comment by yqrashawn** [comment]
* [ ] delete pinned message when don't have perms to unpin the message https://github.com/status-im/status-mobile/issues/14488
* [ ] undo deletion toast UI (will implement this in separate PR) https://github.com/status-im/status-mobile/pull/14376
* [ ] deleted by xxx UI, may need to add `deleted-by` in status-go https://github.com/status-im/status-mobile/issues/14489
* [ ] check code in this PR, see if anything should be moved to status-im2
